datumaro/plugins/data_formats/synthia/format.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from collections import OrderedDict
+from enum import IntEnum
+
+
+class SynthiaFormatType(IntEnum):
+    synthia_rand = 0
+    synthia_sf = 1
+    synthia_al = 2
+
+
+class SynthiaRandPath:
+    IMAGES_DIR = "RGB"
+    LABELS_SEGM_DIR = "GTTXT"
+    SEMANTIC_SEGM_DIR = "GT"
+
+    @classmethod
+    def meta_folders(cls):
+        return [val for val in vars(cls).values() if isinstance(val, str) and "datumaro" not in val]
+
+
+class SynthiaSfPath:
+    IMAGES_DIR = "RGBLeft"
+    SEMANTIC_SEGM_DIR = "GTLeft"
+    DEPTH_DIR = "DepthLeft"
+
+    @classmethod
+    def meta_folders(cls):
+        return [val for val in vars(cls).values() if isinstance(val, str) and "datumaro" not in val]
+
+
+class SynthiaAlPath:
+    IMAGES_DIR = "RGB"
+    SEMANTIC_SEGM_DIR = "SemSeg"
+    DEPTH_DIR = "Depth"
+
+    @classmethod
+    def meta_folders(cls):
+        return [val for val in vars(cls).values() if isinstance(val, str) and "datumaro" not in val]
+
+
+SynthiaRandLabelMap = OrderedDict(
+    [
+        ("Void", (0, 0, 0)),
+        ("Sky", (128, 128, 128)),
+        ("Building", (128, 0, 0)),
+        ("Road", (128, 64, 128)),
+        ("Sidewalk", (0, 0, 192)),
+        ("Fence", (64, 64, 128)),
+        ("Vegetation", (128, 128, 0)),
+        ("Pole", (192, 192, 128)),
+        ("Car", (64, 0, 128)),
+        ("Sign", (192, 128, 128)),
+        ("Pedestrian", (64, 64, 0)),
+        ("Cyclist", (0, 128, 192)),
+        ("Lanemarking", (0, 175, 0)),
+        ("TrafficLight", (0, 128, 128)),
+    ]
+)
+
+
+SynthiaSfLabelMap = OrderedDict(
+    [
+        ("Void", (0, 0, 0)),
+        ("Road", (128, 64, 128)),
+        ("Sidewalk", (244, 35, 232)),
+        ("Building", (70, 70, 70)),
+        ("Wall", (102, 102, 156)),
+        ("Fence", (190, 153, 153)),
+        ("Pole", (153, 153, 153)),
+        ("TrafficLight", (250, 170, 30)),
+        ("TrafficSign", (220, 220, 0)),
+        ("Vegetation", (107, 142, 35)),
+        ("Terrian", (152, 251, 152)),
+        ("Sky", (70, 130, 180)),
+        ("Person", (220, 20, 60)),
+        ("Rider", (255, 0, 0)),
+        ("Car", (0, 0, 142)),
+        ("Truck", (0, 80, 100)),
+        ("Bus", (0, 60, 100)),
+        ("Train", (0, 80, 100)),
+        ("Motorcycle", (0, 0, 230)),
+        ("Bicycle", (119, 11, 31)),
+        ("RoadLines", (157, 234, 50)),
+        ("Other", (72, 0, 98)),
+        ("RoadWorks", (167, 106, 29)),
+    ]
+)
```

Comment:
Here is the label map for Synthia-Rand, while there are some undeclared pixels in data as shown below.
![image](https://user-images.githubusercontent.com/89109581/236149436-50519cf8-69d5-44c7-b915-b863173f1fd7.png)

Reply:
So my suggestion is map them into label 0 for a while. And we may need to specify the default class id into the mask tools,

Reply:
It is suspicious to happen when some files are corrupted due to network issues while downloading it. However, yes, we can introduce `default_id: Optional[int]` argument to `unpaint_mask()` function.
https://github.com/openvinotoolkit/datumaro/blob/98298beaae9f5805f1a51e6d213f0b5f74a7aacf/datumaro/util/mask_tools.py#L56-L87

if `default_id=None`, it will raise an error as is, otherwise it will give warning and assign `default_id`, but no error raise.

Reply:
good, I updated `unpaint_mask` and `load_mask` with `default_id`.